girder_worker/plugins/docker/tests/docker_integration_test.py

```diff
@@ -0,0 +1,278 @@
+import ConfigParser
+import girder_worker
+import os
+import shutil
+import six
+import stat
+import sys
+import unittest
+
+from girder_worker.core import run, io
+
+test_image = 'cjh1/girder_worker:test'
+
+
+def setUpModule():
+    global _tmp
+    _tmp = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), '_tmp', 'docker')
+    girder_worker.config.set('girder_worker', 'tmp_root', _tmp)
+    try:
+        girder_worker.config.add_section('docker')
+    except ConfigParser.DuplicateSectionError:
+        pass
+
+    if os.path.isdir(_tmp):
+        shutil.rmtree(_tmp)
+
+
+def tearDownModule():
+    if os.path.isdir(_tmp):
+        shutil.rmtree(_tmp)
+
+
+class TestDockerMode(unittest.TestCase):
+    """
+    Integration tests that call out to docker rather than using mocks.
+    """
+
+    def setUp(self):
+        self._test_message = 'Hello from girder_worker!'
+        self._tmp = os.path.join(_tmp, 'testing')
+        if not os.path.isdir(self._tmp):
+            os.makedirs(self._tmp)
+
+    def tearDown(self):
+        shutil.rmtree(self._tmp)
+
+    def testDockerModeStdio(self):
+        """
+        Test writing to stdout.
+        """
+
+        task = {
+            'mode': 'docker',
+            'docker_image': test_image,
+            'pull_image': True,
+            'container_args': ['$input{test_mode}', '$input{message}'],
+            'inputs': [{
+                'id': 'test_mode',
+                'name': '',
+                'format': 'string',
+                'type': 'string'
+            }, {
+                'id': 'message',
+                'name': '',
+                'format': 'string',
+                'type': 'string'
+            }],
+            'outputs': []
+        }
+
+        inputs = {
+            'test_mode': {
+                'format': 'string',
+                'data': 'stdio'
+            },
+            'message': {
+                'format': 'string',
+                'data': self._test_message
+            }
+        }
+
+        _old = sys.stdout
+        mockedStdOut = six.StringIO()
```

Comment:
AFP.  Can you change the name of these objects from 'mocked' to maybe 'patched' or 'intercepted'? Unless you feel that this is actually mocking.